File: frontend/src/api/targetSummary.ts
// Display-only summary; no invented ARNs, arbitrary parameter values or per-row raw fetches.
export function targetSummary(record: Record<string, unknown>): string {
  const req = record.requestParameters;
  const p = req && typeof req === "object" && !Array.isArray(req) ? req as Record<string, unknown> : {};
  const text = (v: unknown) => typeof v === "string" ? v : "";
  const resources = Array.isArray(record.resources) ? record.resources : [];
  const first = resources[0] && typeof resources[0] === "object" ? resources[0] as Record<string, unknown> : {};
  const instances = p.instancesSet && typeof p.instancesSet === "object" ? p.instancesSet as Record<string, unknown> : {};
  const instance = Array.isArray(instances.items) && instances.items[0] && typeof instances.items[0] === "object" ? instances.items[0] as Record<string, unknown> : {};
  return (text(p.secretId) || (text(p.bucketName) ? text(p.bucketName) + (text(p.key) ? "/" + text(p.key) : "") : "") || text(first.ARN)
    || text(p.roleArn) || text(p.roleName) || text(p.policyArn) || text(p.instanceId) || text(instance.instanceId)
    || text(p.keyId) || text(p.functionName) || text(p.resourceArn) || text(p.tableName)).slice(0, 512);
}

File: internal/store/target_summary.go
package store

// Window-only display projection. Keep raw evidence out of the row transport and
// avoid a storage migration: old saved evidence gains targets on its next read.
// These are bounded recorded strings, not resource correlation keys or filters.
const targetSummarySQL = `left(coalesce(
 CASE WHEN json_type(raw,'$.requestParameters.secretId')='VARCHAR' THEN nullif(json_extract_string(raw,'$.requestParameters.secretId'),'') END,
 CASE WHEN json_type(raw,'$.requestParameters.bucketName')='VARCHAR' THEN nullif(json_extract_string(raw,'$.requestParameters.bucketName'),'') || CASE WHEN json_type(raw,'$.requestParameters.key')='VARCHAR' THEN '/' || json_extract_string(raw,'$.requestParameters.key') ELSE '' END END,
 CASE WHEN json_type(raw,'$.resources[0].ARN')='VARCHAR' THEN nullif(json_extract_string(raw,'$.resources[0].ARN'),'') END,
 CASE WHEN json_type(raw,'$.requestParameters.roleArn')='VARCHAR' THEN nullif(json_extract_string(raw,'$.requestParameters.roleArn'),'') END,
 CASE WHEN json_type(raw,'$.requestParameters.roleName')='VARCHAR' THEN nullif(json_extract_string(raw,'$.requestParameters.roleName'),'') END,
 CASE WHEN json_type(raw,'$.requestParameters.policyArn')='VARCHAR' THEN nullif(json_extract_string(raw,'$.requestParameters.policyArn'),'') END,
 CASE WHEN json_type(raw,'$.requestParameters.instanceId')='VARCHAR' THEN nullif(json_extract_string(raw,'$.requestParameters.instanceId'),'') END,
 CASE WHEN json_type(raw,'$.requestParameters.instancesSet.items[0].instanceId')='VARCHAR' THEN nullif(json_extract_string(raw,'$.requestParameters.instancesSet.items[0].instanceId'),'') END,
 CASE WHEN json_type(raw,'$.requestParameters.keyId')='VARCHAR' THEN nullif(json_extract_string(raw,'$.requestParameters.keyId'),'') END,
 CASE WHEN json_type(raw,'$.requestParameters.functionName')='VARCHAR' THEN nullif(json_extract_string(raw,'$.requestParameters.functionName'),'') END,
 CASE WHEN json_type(raw,'$.requestParameters.resourceArn')='VARCHAR' THEN nullif(json_extract_string(raw,'$.requestParameters.resourceArn'),'') END,
 CASE WHEN json_type(raw,'$.requestParameters.tableName')='VARCHAR' THEN nullif(json_extract_string(raw,'$.requestParameters.tableName'),'') END,
 ''),512)`

const displayCols = pageCols + "," + targetSummarySQL + " AS target"
